src/artemis/parameters/external_parameters.py

```diff
@@ -0,0 +1,199 @@
+from pathlib import Path
+from typing import NamedTuple, Optional, Union
+
+import jsonschema
+from dataclasses_json import DataClassJsonMixin
+
+from artemis.parameters.constants import (
+    EXPERIMENT_DICT,
+    EXPERIMENT_NAMES,
+    EXPERIMENT_TYPES,
+    PARAMETER_VERSION,
+    SIM_BEAMLINE,
+    SIM_INSERTION_PREFIX,
+    SIM_ZOCALO_ENV,
+)
+from artemis.utils import Point3D
+
+
+def default_field(obj):
+    return field(default_factory=lambda: copy.deepcopy(obj))
+
+
+class WrongExperimentParameterSpecification(Exception):
+    pass
+
+
+@dataclass
+class ExternalDetectorParameters(DataClassJsonMixin):
+    current_energy: int = 100
+    exposure_time: float = 0.1
+    directory: str = "/tmp"
+    prefix: str = "file_name"
+    run_number: int = 0
+    detector_distance: float = 100.0
+    omega_start: float = 0.0
+    omega_increment: float = 0.0
+    num_images: int = 2000
+    use_roi_mode: bool = False
+    det_dist_to_beam_converter_path: str = (
+        "src/artemis/devices/unit_tests/test_lookup_table.txt"
+    )
+    detector_size_constants: Optional[str] = "EIGER2_X_16M"
+
+
+@dataclass
+class ExternalISPyBParameters(DataClassJsonMixin):
+    sample_id: Optional[int] = None
+    sample_barcode: Optional[str] = None
+    visit_path: str = ""
+    pixels_per_micron_x: float = 0.0
+    pixels_per_micron_y: float = 0.0
+    # gets stored as 2x2D coords - (x, y) and (x, z). Values in pixels
+    upper_left: NamedTuple = Point3D(x=0, y=0, z=0)
+    position: NamedTuple = Point3D(x=0, y=0, z=0)
+    xtal_snapshots_omega_start: list[str] = default_field(
+        ["test_1_y", "test_2_y", "test_3_y"]
+    )
+    xtal_snapshots_omega_end: list[str] = default_field(
+        ["test_1_y", "test_2_y", "test_3_y"]
+    )
+    transmission: float = 1.0
+    flux: float = 10.0
+    wavelength: float = 0.01
+    beam_size_x: float = 0.1
+    beam_size_y: float = 0.1
+    focal_spot_size_x: float = 0.0
+    focal_spot_size_y: float = 0.0
+    comment: str = "Descriptive comment."
+    resolution: float = 1
+    undulator_gap: float = 1.0
+    synchrotron_mode: Optional[str] = None
+    slit_gap_size_x: float = 0.1
+    slit_gap_size_y: float = 0.1
+
+
+@dataclass
+class ExternalGridScanParameters(DataClassJsonMixin):
+    x_steps: int = 4
+    y_steps: int = 200
+    z_steps: int = 61
+    x_step_size: float = 0.1
+    y_step_size: float = 0.1
+    z_step_size: float = 0.1
+    dwell_time: float = 0.2
+    x_start: float = 0.0
+    y1_start: float = 0.0
+    y2_start: float = 0.0
+    z1_start: float = 0.0
+    z2_start: float = 0.0
+
+
+@dataclass
+class ExternalRotationScanParameters(DataClassJsonMixin):
+    x_steps: int = 4
+    y_steps: int = 200
+    z_steps: int = 61
+    x_step_size: float = 0.1
+    y_step_size: float = 0.1
+    z_step_size: float = 0.1
+    dwell_time: float = 0.2
+    x_start: float = 0.0
+    y1_start: float = 0.0
+    y2_start: float = 0.0
+    z1_start: float = 0.0
+    z2_start: float = 0.0
```

Comment:
Should: These are just copying the Grid ones? Either put in our best guess or a TODO

Reply:
this is already done in the next part